core/build.go

```diff
@@ -231,6 +250,35 @@ func (b *Build) Run() error {
 	return nil
 }
 
+func (b *Build) preProcessing() error {
+	for _, p := range b.Plugins {
+		prePostPlugin, ok := p.(plugin.PrePostProcessPlugin)
+		if !ok {
+			continue
```

Comment:
We may panic here since `!ok` results from a programming error.

Reply:
No, because I extra created a new Interface, so that the programmer of plugins can decide if he needs this pre/post functionallity.
If not, he can just implement the Plugin interface instead of PrePostPlugin in which case this would result in !ok and just not call the pre-post methods.

That's mainly for backward compatibillity, but also for cleaner plugins, which only include the needed methods.